alica_engine/src/engine/RuleBook.cpp

```diff
@@ -103,6 +103,9 @@ PlanChange RuleBook::visit(RunningPlan& r)
             doDynAlloc = false;
         }
         changeRecord = updateChange(changeRecord, planAbortRule(r));
+        if (changeRecord == PlanChange::FailChange) {
```

Comment:
does a similar check need to be done for the other updateChange calls?

Reply:
No, not at the moment. Though I don't particular like the current setup.
A more thorough refactoring might be in order in the future.